### PR TITLE
Revert "Fixed apparent type of homomorphic mapped type with non-homomorphic instantiation (#56727)" and add test

### PR DESCRIPTION
**Pull request by gabritto**
#56727 breaks @types/sinon version 10: https://github.com/microsoft/TypeScript/pull/56727#issuecomment-1886338688
This PR reverts that one, and adds a test with a simplified version of the code that broke (test thanks to @Andarist).

### CODE REVIEW

**Review by jakebailey — APPROVED** [review]

File: tests/cases/compiler/homomorphicMappedTypeWithNonHomomorphicInstantiationSpreadable1.ts
Comment (jakebailey) [inline tests/cases/compiler/homomorphicMappedTypeWithNonHomomorphicInstantiationSpreadable1.ts:1]: Perhaps leave this test in?

Comment (Andarist) [reply tests/cases/compiler/homomorphicMappedTypeWithNonHomomorphicInstantiationSpreadable1.ts:1]: Isn't it confusing to have tests that are kinda meant to pass but they fail?

**Review by ahejlsberg — APPROVED** [review]

File: tests/cases/compiler/homomorphicMappedTypeWithNonHomomorphicInstantiationSpreadable1.ts
Comment (DanielRosenwasser) [reply tests/cases/compiler/homomorphicMappedTypeWithNonHomomorphicInstantiationSpreadable1.ts:1]: Well the fact that the test was added meant that the behavior wasn't tested in the first place. Capturing the behavior and changes over time is useful.

Comment (gabritto) [reply tests/cases/compiler/homomorphicMappedTypeWithNonHomomorphicInstantiationSpreadable1.ts:1]: I'll remember to leave the tests in the future.

**Pull request merged**